x-pack/plugins/observability_solution/observability_shared/public/locators/infra/inventory_locator.ts

```diff
@@ -0,0 +1,92 @@
+
+export interface InventoryLocatorParams extends SerializableRecord {
+  inventoryViewId?: string;
+  waffleFilter?: {
+    expression: string;
+    kind: string;
+  };
+  waffleTime?: {
+    currentTime: number;
+    isAutoReloading: boolean;
+  };
+  waffleOptions?: {
+    accountId: string;
+    autoBounds: boolean;
+    boundsOverride: {
+      max: number;
+      min: number;
+    };
+  };
+  customMetrics?: string; // encoded value
+  customOptions?: string; // encoded value
+  groupBy?: { field: string };
+  legend?: {
+    palette: string;
+    reverseColors: boolean;
+    steps: number;
+  };
+  metric: string; // encoded value
+  nodeType: string;
+  region?: string;
+  sort: {
+    by: string;
+    direction: 'desc' | 'async';
+  };
+  timelineOpen: boolean;
+  view: 'map' | 'table';
+  state?: SerializableRecord;
+}
+
+export const INVENTORY_LOCATOR_ID = 'INVENTORY_LOCATOR';
+
+export class InventoryLocatorDefinition implements LocatorDefinition<InventoryLocatorParams> {
+  public readonly id = INVENTORY_LOCATOR_ID;
+
+  public readonly getLocation = async (params: InventoryLocatorParams) => {
+    const paramsWithDefaults = {
+      waffleFilter: rison.encodeUnknown(params.waffleFilter ?? { kind: 'kuery', expression: '' }),
+      waffleTime: rison.encodeUnknown(
+        params.waffleTime ?? {
+          currentTime: new Date().getTime(),
+          isAutoReloading: false,
+        }
+      ),
+      waffleOptions: rison.encodeUnknown(
+        params.waffleOptions ?? {
+          accountId: '',
+          autoBounds: true,
+          boundsOverride: { max: 1, min: 0 },
+        }
+      ),
+      customMetrics: params.customMetrics || rison.encodeUnknown(''),
+      customOptions: params.customOptions || rison.encodeUnknown(''),
+      groupBy: rison.encodeUnknown(params.groupBy ?? {}),
+      legend: rison.encodeUnknown(
+        params.legend ?? { palette: 'cool', reverseColors: false, steps: 10 }
+      ),
+      metric: params.metric,
+      nodeType: rison.encodeUnknown(params.nodeType),
+      region: rison.encodeUnknown(params.region ?? ''),
+      sort: rison.encodeUnknown(params.sort ?? { by: 'name', direction: 'desc' }),
+      timelineOpen: rison.encodeUnknown(params.timelineOpen ?? false),
+      view: rison.encodeUnknown(params.view ?? 'map'),
+    };
+
+    return {
+      app: 'metrics',
+      path: `/inventory?waffleFilter=${paramsWithDefaults.waffleFilter}&waffleTime=${paramsWithDefaults.waffleTime}&waffleOptions=${paramsWithDefaults.waffleOptions}&customMetrics=${paramsWithDefaults.customMetrics}&customOptions=${paramsWithDefaults.customOptions}&groupBy=${paramsWithDefaults.groupBy}&legend=${paramsWithDefaults.legend}&metric=${paramsWithDefaults.metric}&nodeType=${paramsWithDefaults.nodeType}&region=${paramsWithDefaults.region}&sort=${paramsWithDefaults.sort}&timelineOpen=${paramsWithDefaults.timelineOpen}&view=${paramsWithDefaults.view}`,
+      state: params.state ? params.state : {},
+    };
```

Comment:
would this work?

```suggestion
    const queryStringParams = queryString.stringify(paramsWithDefaults);
    return {
      app: 'metrics',
      path: `/inventory?${queryStringParams}`,
      state: params.state ? params.state : {},
    };
```

Reply:
Good idea, thanks 👍 I started with just 3 params then decided to support more in case we need them 😅